cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -548,18 +619,18 @@ static PyObject *register_value_change_callback(PyObject *self, PyObject *args)
 static PyObject *iterate(PyObject *self, PyObject *args)
 {
     COCOTB_UNUSED(self);
-    gpi_sim_hdl hdl;
+    gpi_hdl_Object<gpi_sim_hdl> *hdl_obj;
     int type;
     gpi_iterator_hdl result;
     PyObject *res;
 
-    if (!PyArg_ParseTuple(args, "O&i", gpi_sim_hdl_converter, &hdl, &type)) {
+    if (!PyArg_ParseTuple(args, "O!i", &std::remove_pointer<decltype(hdl_obj)>::type::py_type, &hdl_obj, &type)) {
```

Comment:
Yuck, C++ does have a tendency to get ugly. In this case type reflection isn't necessary. My rule of thumb is use it when you are writing generic code or when it is more concise/readable.
```suggestion
    if (!PyArg_ParseTuple(args, "O!i", &gpi_hdl_Object<gpi_sim_hdl>::py_type, &hdl_obj, &type)) {
```

Reply:
I'd argue it is necessary, because a typo here would not be detected at compile-time.

Reply:
I've made it shorter anyway